python/remote-control-center/README.md

```diff
@@ -0,0 +1,123 @@
+
+## Initial Deployment
+This solution leverages the AWS Serverless Application Model to deploy the backend infrastructure required to implement the solution.
+
+### Deploy the solution
+1. Clone this directory
+2. Ensure the AWS CLI and AWS SAM are installed locally (or leverage a Cloud9 development environment)
+   1. https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html
+3. Run the following commands
+   1. ` sam build `
+   2. ` sam package --s3-bucket <YOUR_S3_BUCKET> --output-template-file packaged.yaml `
+   3. ` sam deploy --template-file packaged.yaml --stack-name remote-control-center   --capabilities CAPABILITY_IAM  `
+4. Wait while the template deploys the following resources
+   1. ConfigTable - The Amazon DynamoDB table used to store configurations
+   2. HydrateConfigTableLambda - An AWS Lambda function used to populate values into the config table
+   3. RetrieveConfigLambda - An AWS Lambda function to be invoked from within a contact flow
+   4. HydrateConfigTableLambdaRole - An AWS IAM Role for HydrateConfigTableLambda
+   5. RetrieveConfigLambdaRole - An AWS IAM Role for RetrieveConfigLambda
+
+### Configure example configurations
+1. Navigate to the HydrateConfigTableLambda function
+2. Included in the function package is a configs.json file with example configurations
+3. The function code includes comments on how to create additional configurations using events
+4. Create and execute a test event with the following JSON body:
+```javascript
+{
+  "LanguageCodes": [
+    "en",
+    "fr",
+    "es"
+  ]
+}
+```
+5. This will populate the table with the default configurations and translate messages to French and Spanish
+6. Navigate to the ConfigTable in DynamoDB to validate the configurations were properly loaded
+
+### Configure Amazon Connect to leverage the Control Center
+1. Navigate to the Amazon Connect console
+2. Select your instance (or create a new one)
+3. Within the Amazon Connect service page under Contact Flows, give your instance permission to the RetrieveConfigLambda function
+4. Log into the instance
+5. Under routing, select contact flows and select create new contact flow
+6. Import the example remote-Sample.json contact flow
+7. For the two Invoke AWS Lambda function blocks, update the target lambda function to the RetrieveConfigLambda function
+8. Save, publish, and route a phone number to the contact flow
+9. Test the contact flow for each of the available languages.  After testing option 4 (Italian) notice how the prompts were automatically translated into Italian and then stored in the DynamoDB table
+10. Set the HOT_MESSAGE_FLAG configuration to 1 and test.  Then set it to 2
```

Comment:
I assume this flag is set in the contact flow?

Reply:
Added information on how to change the item in the dynamodb table:
"10. Navigate to the Amazon DynamoDB service and select the ConfigTable.  Edit the item for the HOT_MESSAGE_FLAG configuration and set it to 1, save the item, and test.  Then try setting it to 2.  Notice how the behavior of the contact flow changed without having to make any modifications to the contact flow itself.  When the flag is equal to 1, the HOT_MESSAGE configuration is played out before entering the main menu.  When the flag is set to 2, the EMERGENCY_MESSAGE configuration is played out, and the caller is disconnected."